internal/cli/global.go

```diff
@@ -70,17 +79,17 @@ func (o *GlobalOptions) WithTimeout(ctx context.Context) (context.Context, conte
 	return ctx, func() {}
 }
 
-func ConfigFilePath(context string) string {
-	if len(context) > 0 && context != "default" {
-		return filepath.Join(ConfigDir(), defaultConfigFileName+"_"+context+"."+defaultConfigFileExt)
+func (o *GlobalOptions) determineConfigDir() string {
+	if o.ConfigDirOverride != "" {
+		return o.ConfigDirOverride
 	}
-	return filepath.Join(ConfigDir(), defaultConfigFileName+"."+defaultConfigFileExt)
+	return filepath.Join(homedir.HomeDir(), ".config", "flightctl")
```

Comment:
Note that `$HOME/.config/flightctl` is only correct on Linux and even only on _some_ distros. To ensure it works correctly everywhere (incl. Darwin and Windows), we need to keep using `os.UserConfigDir` instead.

Nit: Changing the order and naming of these functions to get the config dir and file path (instead of just turning them into a method) makes the diff unnecessarily harder to review.